src/Compilers/Core/Portable/PublicAPI.Shipped.txt

```diff
@@ -2210,7 +2210,6 @@ Microsoft.CodeAnalysis.SyntaxTokenList.Reverse() -> Microsoft.CodeAnalysis.Synta
 Microsoft.CodeAnalysis.SyntaxTokenList.Reversed
 Microsoft.CodeAnalysis.SyntaxTokenList.Reversed.Enumerator
 Microsoft.CodeAnalysis.SyntaxTokenList.Reversed.Enumerator.Current.get -> Microsoft.CodeAnalysis.SyntaxToken
-Microsoft.CodeAnalysis.SyntaxTokenList.Reversed.Enumerator.Enumerator(ref Microsoft.CodeAnalysis.SyntaxTokenList list) -> void
```

Comment:
Breaking change?

Reply:
Technically yes. But I am certain that it was public by mistake. Noone should be instantiating these enumerators except via `GetEnumerator` call. No other struct enumerator have a public constructor. Especially those that take byre parameters.

We can discuss this further but while most compat breaks are bad, not all are. I think this public exposure should be undone ASAP. - not just because it stands in the way of changing from `ref` -> `in`, I could leave it as `ref`. I left `ref` in other public places, for compat reasons. 

Here a public ctor just makes no sense.

---
In reply to: [180876819](https://github.com/dotnet/roslyn/pull/26010#discussion_r180876819) [](ancestors = 180876819)

Reply:
I am simply removing the ctors. It is not related to `in` vs. `ref`

---
In reply to: [180885818](https://github.com/dotnet/roslyn/pull/26010#discussion_r180885818) [](ancestors = 180885818)

Reply:
I have sent a note to Compat Council to decide on this, but I think we should remove these from the public API.

I can keep the `ref` ctors public (for what use?) and add `in` ctor with a dummy parameter for the use in GetEnumerator - to avoid defensive copying, but it would be a very counterproductive "fix"

---
In reply to: [180900004](https://github.com/dotnet/roslyn/pull/26010#discussion_r180900004) [](ancestors = 180900004,180876819)

Reply:
Compat Council is ok with removing the following from the public surface:

1. Microsoft.CodeAnalysis.SyntaxTokenList.Reversed.Enumerator.Enumerator(ref Microsoft.CodeAnalysis.SyntaxTokenList list)

2. Microsoft.CodeAnalysis.SyntaxTriviaList.Reversed.Enumerator.Enumerator(ref Microsoft.CodeAnalysis.SyntaxTriviaList list)



---
In reply to: [180905813](https://github.com/dotnet/roslyn/pull/26010#discussion_r180905813) [](ancestors = 180905813,180900004,180876819)